### ISSUE TITLE
Some questions about weights file

### ISSUE THREAD

**Issue by cfanyyx** [mystic123/tensorflow-yolo-v3#38]
Thanks for your nice codes, and I have two questions about the yolo weight file:
1. In your load_weight function, you read the weights in the order of 'gamma, beta, mean, var', but then you change the order to 'beta, gamma, mean, var' in next line, is that correct?
===========================================================
if 'BatchNorm' in var2.name.split('/')[-2]:
    gamma, beta, mean, var = var_list[i + 1:i + 5]
    batch_norm_vars = [beta, gamma, mean, var]
    for var in batch_norm_vars:
        shape = var.shape.as_list()
        num_params = np.prod(shape)
        var_weights = weights[ptr:ptr + num_params].reshape(shape)
        ptr += num_params
        assign_ops.append(tf.assign(var, var_weights, validate_shape=True))
===========================================================

2. I find there are only 4 variables in the head of weight file in save_weight_up_to function in https://github.com/pjreddie/darknet/blob/master/src/parser.c, but you mention 5 in your readme article https://itnext.io/implementing-yolo-v3-in-tensorflow-tf-slim-c3c55ff59dbe, and  I find only 'scale, rolling_mean, rolling_variance' in save_batchnorm_weights function. So I am confused about the total weight file. Can you help me out?

**Comment by mystic123** [comment]
Hi cfanyyx,

Good questions. Here are the answers:
1) It is because the order of variables in tensorflow is different than in darknet project. Because of this we also need to change order of variables of convolutional layers in line 298 of load_weights function
2) You are right, there are 4 variables, but the last one of them is of type size_t which usually is 64bit integer. So total number of bytes to skip equals (int32 + int32 + int32 + int64 = 5 * int32). I should have mentioned that.

**Comment by cfanyyx** [comment]
Thanks for your reply! There remains a question that I don't understand. In the save_batchnorm_weights() function in https://github.com/pjreddie/darknet/blob/master/src/parser.c, there are only scales, rolling_mean, rolling_variance, I am curious that how they correspond to _gamma, beta, mean, var_ these four parameters?
===============================
void save_batchnorm_weights(layer l, FILE *fp) {
.......
fwrite(l.scales, sizeof(float), l.c, fp);
fwrite(l.rolling_mean, sizeof(float), l.c, fp);
fwrite(l.rolling_variance, sizeof(float), l.c, fp);
}
===============================
